src/lightning/fabric/accelerators/cuda.py

```diff
@@ -90,6 +90,8 @@ def find_usable_cuda_devices(num_devices: int = -1) -> List[int]:
         both processes determine that the device is unoccupied, leading into one of them crashing later on.
 
     """
+    if num_devices == 0:
+        raise ValueError("`num_devices=0` is not a valid number of devices to look for. Pass a positive number.")
```

Comment:
One could argue that this should return an empty list instead. If there are valid reasons to call `find_usable_cuda_devices(0)`, then returning an empty list would make the user's code nicer than having to catch this ValueError